components/director/internal/domain/runtime/service.go

```diff
@@ -34,6 +36,7 @@ type LabelRepository interface {
 	ListForObject(ctx context.Context, tenant string, objectType model.LabelableObject, objectID string) (map[string]*model.Label, error)
 	Delete(ctx context.Context, tenant string, objectType model.LabelableObject, objectID string, key string) error
 	DeleteAll(ctx context.Context, tenant string, objectType model.LabelableObject, objectID string) error
+	DeleteByKeyNotPattern(ctx context.Context, tenant string, objectType model.LabelableObject, objectID string, labelKeyPattern string) error
```

Comment:
Wonder if we can get an easier to understand name - it took me like 5-10 seconds by just looking at the interface (without seeking the implmentation).
Maybe - `DeleteByKeyNegationPattern`?